actions.py

```diff
@@ -261,14 +268,21 @@ def get_duration(entity):
                 deltaargs = {f"{grain}s": 1}
                 delta = relativedelta.relativedelta(**deltaargs)
                 parsedend = parsedstart + delta
+                reportgrain = grain
+
+            if any(grain == t for t in ["day","week","month","year"]):
+                dateformat = "%A %b %d, %Y"
+            else:
+                dateformat = "%H:%M %A %b %d, %Y"
+
+            strstart = parsedstart.strftime(dateformat)
+            strend = parsedend.strftime(dateformat)
```

Comment:
can we rename these variables to something more descriptive?

Reply:
yup